pydynamo_brain/pydynamo_brain/ui/actions/radiiActions.py

```diff
@@ -24,7 +35,92 @@ def changeRadius(self, localIdx, dR, laterStacks=False):
             for atIndex in range(localIdx, upperBound):
                 current = self._localState(atIndex).currentPoint()
                 if current is not None:
+                    current.manuallyMarked = False
                     if current.radius == None:
                         current.radius = 10 * dR
                     else:
                         current.radius *= dR
+
+
+def _radiiThreshold(xs, ys):
+    for i, x in enumerate(xs):
+        if ys[i] <= 0.05:
+            return xs[i]
+
+def _somaThreshold(xs, ys):
+    index = ys.index(np.max(ys))
+    return xs[index]
+
+def intensityForPointRadius(volume, point):
+    zIdx = int(point.location[2])
+    if point.isRoot():
+        plane = volume[0, zIdx, :, :]
+        planeMod = roberts(plane)
+        planeMod = ndimage.gaussian_filter(planeMod, sigma=3)
+        edges = feature.canny(plane, sigma=2)
+        edges[edges == 0]= np.nan
+        planeMod = edges * planeMod
+
+    else:
+        plane = volume[0, (zIdx-1):(zIdx+1), :, :]
+        plane = np.amax(plane, axis=0)
+        planeMod = roberts(plane)
+        planeMod = ndimage.gaussian_filter(planeMod, sigma=3)
+        edges = feature.canny(plane, sigma=2)
+        planeMod = planeMod - edges
+
+    plane01 = np.ones(plane.shape)
+    plane01[int(point.location[1]), int(point.location[0])] = 0
+    planeDist = ndimage.distance_transform_edt(plane01)
+
+    _mx = np.max(planeDist)
+    planeDistNorm = np.power(((_mx - planeDist) / _mx), 13)
+
+    X_POINTS = 100
+    SQUISH = 1
+    MAX_DIST_PX = 30
+
+    xs = 1 + np.power(np.arange(0, 1, 1 / X_POINTS), SQUISH) * (MAX_DIST_PX - 1)
+    ys = []
+    for x in xs:
+        selected = (planeDist < x)
+        ys.append(np.mean(planeMod[selected]))
+
+    if point.isRoot():
+        radius = _somaThreshold(xs, ys)
+    else:
+        radius = _radiiThreshold(xs, ys)
+
+
+    #prevent terminal points from having a radius larger than their parent
+    if point.isLastInBranch() and (point.isRoot()==False):
+        parentPoint = point.pathFromRoot()[-2]
+        parentRadius = parentPoint.radius
+        if parentRadius == None:
+            parentRadius = 1
+        if radius == None:
+                radius = 1
+        if radius >= parentRadius:
+            radius = parentRadius
```

Comment:
maybe with some buffer? e.g. `if radius >= parentRadius * 2` or whatever
in theory they can get bigger.....